src/ucp/wireup/select.c

```diff
@@ -769,6 +798,53 @@ static ucs_status_t ucp_wireup_add_am_lane(ucp_ep_h ep, const ucp_ep_params_t *p
     return UCS_OK;
 }
 
+static ucs_status_t ucp_wireup_add_rma_bw_lanes(ucp_ep_h ep,
+                                                const ucp_ep_params_t *params,
+                                                unsigned address_count,
+                                                const ucp_address_entry_t *address_list,
+                                                ucp_wireup_lane_desc_t *lane_descs,
+                                                ucp_lane_index_t *num_lanes_p)
+{
+    ucp_wireup_criteria_t criteria;
+    ucs_status_t status;
+
+    if (ucp_ep_get_context_features(ep) & UCP_FEATURE_RMA) {
+        /* if needed for RMA, need also access for remote allocated memory */
+        criteria.remote_md_flags = criteria.local_md_flags = 0;
+    } else {
+        if (!(ucp_ep_get_context_features(ep) & UCP_FEATURE_TAG)) {
+            return UCS_OK;
+        }
+        /* if needed for RNDV, need only access for remote registered memory */
+        criteria.remote_md_flags = criteria.local_md_flags = UCT_MD_FLAG_REG;
+    }
+
+    criteria.title              = "high-bw remote %s memory access";
+    criteria.local_md_flags     = 0;
+    criteria.remote_md_flags    = 0;
+    criteria.remote_iface_flags = UCT_IFACE_FLAG_GET_ZCOPY |
+                                  UCT_IFACE_FLAG_PUT_ZCOPY;
+    criteria.local_iface_flags  = criteria.remote_iface_flags |
+                                  UCT_IFACE_FLAG_PENDING;
+    criteria.calc_score         = ucp_wireup_rma_bw_score_func;
+    criteria.tl_rsc_flags       = 0;
+    ucp_wireup_fill_ep_params_criteria(&criteria, params);
+
+    if (ucs_test_all_flags(ucp_ep_get_context_features(ep),
+                           UCP_FEATURE_TAG|UCP_FEATURE_WAKEUP)) {
+        criteria.local_iface_flags |= UCP_WORKER_UCT_UNSIG_EVENT_CAP_FLAGS;
+    }
+
+    /* TODO enable selecting more than one lane on a device by passing maximal
+     * number of lanes to use to this function */
+    status = ucp_wireup_add_memaccess_lanes(ep, address_count, address_list,
```

Comment:
is it ok that ucp_wireup_add_memaccess_lanes selects lanes with max scores only? for case when 2 ports have minimal diff by score it will select only one

Reply:
@hoopoepg this PR does not have multi-rail support, i assumed you could rebase your MR code on on.